src/js/server/devpmt-express.js

```diff
@@ -0,0 +1,528 @@
+fluid.require("gpii-express");
+fluid.require("gpii-express-user");
+fluid.require("gpii-handlebars");
+require("gpii-universal");
+
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.handlebars");
+
+
+/**
+ * gpii.handlebars.requestFuncTransform
+ *  This component is sort of a facsimile of the functions available
+ *  in many web frameworks that take a request instance or URL parameter
+ *  and return something for the response. In this case what is returned
+ *  will be assigned to the variable in the transform.
+ */
+fluid.defaults("gpii.handlebars.requestFuncTransform", {
+    gradeNames: ["fluid.component"]
+});
+
+gpii.handlebars.requestFuncTransform = function (value, transformSpec) {
+    return fluid.invokeGlobalFunction(transformSpec.func, [transformSpec.that, value]);
+};
+
+/**
+ * Allows fetching a component from the tree using an IoC expression of the
+ * type one would normally place in a components defaults block. If there is
+ * more than one instance of this component, will use the first instance in
+ * the returned list.
+ *
+ * An example path might be:
+ * "{gpii.devpmt}.options.prefSetDir"
+ *
+ * but could be any resolvable path on the component.
+ *
+ * @param {String} path - The IoC path to look up.
+ * @return {Any} The value at the path.
+ */
+fluid.getValueByGlobalPath = function (path) {
+    var contextRef = fluid.parseContextReference(path);
+    var comp = fluid.queryIoCSelector(fluid.rootComponent, contextRef.context)[0];
+    return fluid.getForComponent(comp, contextRef.path);
+};
+
+/**
+ * gpii.devpmt.npset - Infusion component representing a single Preferences Set for
+ * a user or snapset. Based on current hardwire, loads it's Preferences Set from
+ * the bundled GPII's testData.
+ *
+ * When creating a new instance, the prefsSafeName is required in the options
+ * block.
+ *
+ * gpii.devpmt.npset({ prefsSafeName: "alice" , flatPrefs: flatPrefsObject});
+ *
+ * This will become a model component for Preferences set editing.
+ */
+fluid.defaults("gpii.devpmt.npset", {
+    gradeNames: ["fluid.component"],
+    // prefsSafeName: "elod",
+    // flatPrefs: {},
+    // docs: {}
+    invokers: {
+        contextNames: {
+            funcName: "gpii.devpmt.contextNames",
+            args: ["{that}.options.flatPrefs"]
+        },
+        prefsSafeApplications: {
+            funcName: "gpii.devpmt.prefsSafeApplications",
+            args: ["{that}.options.flatPrefs"]
+        }
+    }
+});
+
+gpii.devpmt.redisStore = function () {
+    gpii.devpmt.GPII_REDIS_HOST = process.env.GPII_REDIS_HOST || "127.0.0.1";
+    gpii.devpmt.GPII_REDIS_PORT = process.env.GPII_REDIS_PORT || 6379;
+    return new RedisStore({
+        host: gpii.devpmt.GPII_REDIS_HOST,
+        port: gpii.devpmt.GPII_REDIS_PORT
+    });
+};
+
+/**
+ * gpii.devpmt - Main component of the gpii.devpmt server to view and edit
+ * Preferences Sets.
+ */
+gpii.devpmt.LISTEN_PORT = process.env.GPII_DEVPMT_LISTEN_PORT || 8085;
```

Comment:
Pretty weird stuff - would you like to pull these values in by using a config-driven env system of the kind, for example, we use at https://github.com/GPII/universal/blob/master/gpii/configs/gpii.config.preferencesServer.standalone.production.json5#L14

Reply:
I am now pulling this in from the config